src/backend/cuda/JIT/ShiftNode.hpp

```diff
@@ -15,22 +15,20 @@
 
 namespace cuda
 {
-
 namespace JIT
 {
     template<typename T>
     class ShiftNode : public Node
     {
     private:
-
-        Node_ptr m_buffer_node;
+        std::shared_ptr<BufferNode<T>> m_buffer_node;
```

Comment:
I though `Note_ptr` just a typedef for `std::shared_ptr<BufferNode<T>>`.

Reply:
Its a typedef for its base class `std::shared_ptr<Node<T>>`